pkg/monitors/elasticsearch/query/monitor.go

```diff
@@ -0,0 +1,136 @@
+package query
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/signalfx-agent/pkg/core/common/httpclient"
+	"github.com/signalfx/signalfx-agent/pkg/core/config"
+	"github.com/signalfx/signalfx-agent/pkg/monitors"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+	"github.com/signalfx/signalfx-agent/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig  `yaml:",inline" acceptsEndpoints:"true"`
+	httpclient.HTTPConfig `yaml:",inline"`
+
+	Host string `yaml:"host" validate:"required"`
+	Port string `yaml:"port" validate:"required"`
+
+	MetricQueries []*metricQuery `yaml:"metricQueries" validate:"required"`
+}
+
+// Conceptually the data field in search queries define  bucket aggregations
+// that determine "buckets" (some group of elasticsearch documents that match
+// the search query) and metric aggregations that compute metrics from each "bucket"
+type metricQuery struct {
+	// Index that's being queried
+	Index string `yaml:"index" validate:"required"`
+	// Lucene search query which the elasticsearch documents should match
+	LuceneSearchQuery string `yaml:"luceneSearchQuery" validate:"required"`
+	// Query Interval
+	QueryInterval time.Duration `yaml:"queryInterval" default:"30s"`
+	// HTTP Post data field for aggregations
+	HTTPPostData string `yaml:"httpPostData" validate:"required"`
+}
+
+// Monitor for ES queries
+type Monitor struct {
+	Output types.FilteringOutput
+	cancel context.CancelFunc
+	ctx    context.Context
+	logger *utils.ThrottledLogger
+}
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+type metricQueryInfo struct {
+	interval float64
+	aggsMeta map[string]*AggregationMeta
+}
+
+// Configure monitor
+func (m *Monitor) Configure(c *Config) error {
+	m.logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 20*time.Second)
+
+	httpClient, err := c.HTTPConfig.Build()
+	if err != nil {
+		return err
+	}
+	esClient := NewESQueryClient(c.Host, c.Port, c.HTTPConfig.Scheme(), httpClient)
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+
+	metricQueryInfos := make([]*metricQueryInfo, len(c.MetricQueries))
```

Comment:
Looks like this code down to line 88 could be extracted into a function that takes a config and returns a `[]*metricQueryInfo`. A matter of taste so up to you.

Reply:
Ended up removing this logic based on comments from Ben about supporting one query per instance of monitor

Reply:
Yeah, one query can support multiple aggregations so one should be enough.